pkg/config/deployment_ecs.go

```diff
@@ -69,3 +70,28 @@ type ECSPrimaryRolloutStageOptions struct {
 // ECSCanaryCleanStageOptions contains all configurable values for a ECS_CANARY_CLEAN stage.
 type ECSCanaryCleanStageOptions struct {
 }
+
+// ECSTrafficRoutingStageOptions contains all configurable values for ECS_TRAFFIC_ROUTING stage.
+type ECSTrafficRoutingStageOptions struct {
+	// Canary represents the amount of traffic that the rolled out CANARY variant will serve.
+	Canary int `json:"canary"`
+	// Primary represents the amount of traffic that the rolled out CANARY variant will serve.
+	Primary int `json:"primary"`
+}
+
+func (opts ECSTrafficRoutingStageOptions) Percentage() (primary, canary int) {
+	if opts.Primary > 0 && opts.Primary < 100 {
+		primary = opts.Primary
+		canary = 100 - primary
+		return
+	}
+	if opts.Canary > 0 && opts.Canary < 100 {
```

Comment:
`opts.Canary <= 100`
In case `canary = 100, primary = 0`, `primary` should be 0.

Reply:
Nice catch 🙏

Reply:
addressed on https://github.com/pipe-cd/pipe/pull/2072/commits/06a9629150c5bf95fff11da9f3ecef4300098de8 🙏